transport-native-epoll/src/main/java/io/netty/channel/epoll/IovArray.java

```diff
@@ -89,19 +89,31 @@ boolean add(ByteBuf buf) {
 
         final long addr = buf.memoryAddress();
         final int offset = buf.readerIndex();
-        add(addr, offset, len);
-        return true;
+        return add(addr, offset, len);
     }
 
-    private void add(long addr, int offset, int len) {
+    private boolean add(long addr, int offset, int len) {
         if (len == 0) {
             // No need to add an empty buffer.
-            return;
+            return true;
         }
 
         final long baseOffset = memoryAddress(count++);
         final long lengthOffset = baseOffset + ADDRESS_SIZE;
 
+        if (size + len > Integer.MAX_VALUE) {
```

Comment:
Your call. I vote to use appropriate data types for what we support.  I don't think this is part of the public API and so it should be easy to change back in the future if we support `long`.